### DISCUSSION TITLE
Anyone has idea how to insert different font size text lines to customtkinter?

### DISCUSSION THREAD

**Discussion by Gasiro** [TomSchimansky/CustomTkinter#2463] [Q&A] [answered]
Like what I used to do with tkinter: But I can't do it in customtkinter text wight now.
<img width="315" alt="Screenshot 2024-06-11 at 03 47 58" src="https://github.com/TomSchimansky/CustomTkinter/assets/117976338/823c1d3a-ddfb-423c-ba90-b6893743541c">

**Comment by Gasiro** [comment]
This is what I used to do with tkinter:
```
    text_widget = Text(right_frame, wrap="word", width=30, height=10, font=font.Font(size=13), padx=5, pady=5)
    text_widget.pack(fill="both", expand=True)

    now = datetime.now()
    formatted_date_time = now.strftime("%Y/%m/%d %H:%M")
    combined_font = font.Font(size=16, weight="bold")
    text_widget.tag_configure("combined_font", font=combined_font, foreground="chocolate3")
    text_widget.insert("1.0", f"📝 Fitness Rabbit Note","combined_font")
    text_widget.insert("2.0", f"\n{formatted_date_time}")
    text_widget.insert("3.0", f"\n")
    text_widget.insert("4.0", f"\n")
```

**Comment by dipeshSam** [comment]
Use CTkTextbox. A wrapper around Tkinter's Text.

Regards.

**Reply by Gasiro** [reply]
hi Dipesh, this is what I can do now, to change the color in CTkTextbox. However failed to change the individual line to different font size and not able to make individual **bold** or _italic_
`    now = datetime.now()
    formatted_date_time = now.strftime("%Y/%m/%d %H:%M")

    # define different color
    text_widget.tag_config("combined_font", foreground="chocolate3") #蓝色
    
    # insert title and time
    text_widget.insert("1.0", f"📝 Fitness Rabbit Note", "combined_font")
    text_widget.insert("2.0", f"\n{formatted_date_time}")
    text_widget.insert("3.0", f"\n")
    text_widget.insert("4.0", f"\n")`

**Reply by dipeshSam** [reply]
Got it! Let me write an example and elaborate it. Wait a little bit.

Regards.

**Reply by dipeshSam** [reply, marked as answer]
@Gasiro

`customtkinter` prohibits font option in Textbox widget as it considers run time scaling of the window. Scaling for different fonts in customtkinter Textbox is not handled, and if you are sure that user will not scale-up window at runtime time, then you can neglect its prohibition.

What happens is that, customtkinter checks the `font` parameter in `tag_config` method to raise an error. So, instead of passing font parameter directly, **pass it in including as configuration dictionary** (an especial argument provided by Tkinter).


Tkinter provides a parameter `cnf` (stands for configuration dictionary) using which programmar can pass all function arguments as dictionary object.

Sample code:
``` Python

from customtkinter import CTkTextbox, CTkFont, CTk


app = CTk()
app.geometry("500x400")

font1 = CTkFont("Arial", 30)
font2 = CTkFont("impact", 25)
font3 = CTkFont("Georgia", 20)
font4 = CTkFont("Helvetica", 15)

tb = CTkTextbox(app)
tb.place(relx=0.5, anchor="center", rely=0.5, relwidth=1)

tb.insert("1.0", "Text example line 1.\n", "font1")
tb.insert("2.0", "Text example line 2.\n", "font2")
tb.insert("3.0", "Text example line 3.\n", "font3")
tb.insert("4.0", "Text example line 4.", "font4")


tb.tag_config("font1", cnf = {"font": font1})
tb.tag_config("font2", cnf = {"font": font2})
tb.tag_config("font3", cnf = {"font": font3})
tb.tag_config("font4", cnf = {"font": font4})

app.mainloop()
```

Output:
![Screenshot_2024_0611_084933.jpg](https://github.com/TomSchimansky/CustomTkinter/assets/72375763/897998d4-f433-45e3-a7c4-9dbca77921e3)

Hope, it will be helpful for you.
Regards.

**Reply by Gasiro** [reply]
Thank you so much Dipesh! It works now‼️‼️

**Reply by dipeshSam** [reply]
You are welcome @Gasiro.

Best regards.

**Reply by SUR9567** [reply]
hai how to print logos(.png) using print command  in notepad for printing bills and also bigsize fonts are not printed in note pad
